TESTING.md

```diff
@@ -0,0 +1,411 @@
+- `test/test/test_pngcheck_cli.c` - CLI functionality tests (manual)
+- `test/test/test_pngcheck_suite.c` - PNG validation tests (auto-generated)
+
+### Directory layout
+
+```
+pngcheck/
+├── CMakeLists.txt                  # Production build configuration
+├── project.yml                     # Ceedling test configuration
+├── TESTING.md                      # This documentation
+└── test/
+    ├── CMakeLists.txt              # CMake test configuration
+    ├── bin/                        # Ruby CLI tool for managing tests
+    │   └── pngcheck-test           # Ruby script to manage tests and expectations
+    ├── lib/                        # Ruby libraries for `pngcheck-test`
+    ├── test/                       # Test files
+    │   ├── test_pngcheck_cli.c     # CLI tests (manual)
+    │   ├── test_pngcheck_suite.c   # PNG suite tests (generated by test/bin/pngcheck-test)
+    │   └── support/                # Test helpers
+    ├── fixtures/pngsuite/          # PNG test files (176 files)
+    └── expectations/pngsuite/      # Expected outputs
+```
+
+## Test approach
+
+Tests validate pngcheck behavior using **simple prefix matching and exit codes**:
+
+- **Exit code**: 0 for valid PNGs, non-zero for errors
+- **Output prefix**: "OK:" for valid files, "ERROR:" for invalid files
+
+This approach focuses on core functionality rather than exact output matching.
+
+### CLI tests
+
+Static tests for command-line options:
+- Help option (`-h`)
+- File handling (valid, invalid, nonexistent)
+- Output options (`-v`, `-q`, `-t`, `-p`, `-c`, `-7`)
+
+### PNG suite tests
+
+Auto-generated tests using the [PngSuite](http://www.schaik.com/pngsuite/) collection:
+- **Valid PNGs**: Various formats, bit depths, interlacing
+- **Invalid PNGs**: Corrupted headers, bad checksums, truncated files
+
+**Note:** The PngSuite files are committed to the repository at
+`test/fixtures/pngsuite/`, where its LICENSE also resides.
+
+
+## Test management
+
+The test suite includes a Ruby-based management tool.
+
+**Note:** Prerequisites need to be satisfied to use this tool (see
+[prerequisites](#prerequisites)).
+
+To manage tests, run the Ruby CLI tool:
+
+```bash
+# Check status
+bundle exec test/bin/pngcheck-test status
+
+# Regenerate expectations (when pngcheck behavior changes) (add --force to overwrite)
+bundle exec test/bin/pngcheck-test expectations
+
+# Generate test/test/test_pngcheck_suite.c
+bundle exec test/bin/pngcheck-test generate
+```
+
+**Note:**: On Windows the command would be `bundle exec ruby
+test/bin/pngcheck-test` instead of `bundle exec test/bin/pngcheck-test` as the
+Ruby script is not executable on Windows.
+
+## Adding tests
+
+### CLI tests
+Edit `test/test_pngcheck_cli.c` and use helper functions:
+
+```c
+void test_my_feature(void) {
+    test_pngcheck_with_options("-option", "file.png", 0, "expected_content");
+}
+```
+
+### PNG tests
+1. Add PNG files to `fixtures/pngsuite/`
```

Comment:
I assume it would be easy in future to add additional directories under fixtures? I am thinking of the APNG testsuite, and third edition test files, for example.